packages/app/src/files-manager/actions/upload-file.action.ts

```diff
@@ -0,0 +1,64 @@
+/**
+ * Copyright (c) Mudita sp. z o.o. All rights reserved.
+ * For licensing, see https://github.com/mudita/mudita-center/blob/master/LICENSE.md
+ */
+
+import { DeviceType } from "@mudita/pure"
+import { createAsyncThunk } from "@reduxjs/toolkit"
+import { State } from "App/core/constants"
+import { ReduxRootState } from "App/__deprecated__/renderer/store"
+import {
+  FilesManagerEvent,
+  EligibleFormat,
+  DeviceDirectory,
+} from "App/files-manager/constants"
+import { getPathsRequest } from "App/file-system/requests"
+import { uploadFilesRequest } from "App/files-manager/requests"
+import { getFiles } from "App/files-manager/actions/get-files.action"
+import { setUploadingState } from "App/files-manager/actions/base.action"
+
+export const uploadFile = createAsyncThunk(
+  FilesManagerEvent.UploadFiles,
+  async (_, { getState, dispatch, rejectWithValue }) => {
+    const state = getState() as ReduxRootState
+
+    const filesToUpload = await getPathsRequest({
+      filters: [
+        {
+          name: "Audio",
+          extensions: [
+            EligibleFormat.FLAC,
+            EligibleFormat.MP3,
+            EligibleFormat.WAV,
+          ],
+        },
+      ],
+      properties: ["openFile", "multiSelections"],
+    })
+
+    if (!filesToUpload.ok || !filesToUpload.data) {
+      return rejectWithValue(filesToUpload.error)
+    }
+
+    dispatch(setUploadingState(State.Loading))
```

Comment:
Shouldn't it go at the beginning of this action?

Reply:
Nope, because the previous action opens a system popup (for selecting files). During this action don't want to show the loading modal only when the action uploading process starts.